syntax = "proto3";

import "AHRS.proto";
import "pressureTransducers.proto";
import "hydrophones.proto";
import "DVL.proto";
import "leakDetect.proto";
//import "thrusterFeedback.proto"; TODO
import "PMUD.proto";
import "pneumatics.proto";
import "guiComm.proto";

enum Type {
    AHRS_DATA            = 0;
    PRESSURE_TRANSDUCERS = 1;
    HYDROPHONES          = 2;
    DVL_DATA             = 3;
    LEAK_DETECTION       = 4;
    THRUSTER_FEEDBACK    = 5;
    PMUD_DATA            = 6;
    PNEUMATICS           = 7;
    GUI_COMM             = 8;
}

message Mechatronics {
    Type type                         = 1;
    AHRS ahrs                         = 2;
    PressureTransducers pressureTrans = 3;
    Hydrophones hydros                = 4;
    DVL dvl                           = 5;
    LeakDetect leakDetect             = 6;
//    ThrusterFeedback thrusterFeedback = 7; TODO
    PMUD pmud                         = 8;
    Pneumatics pneumatics             = 9;
    GUIComm guiComm                   = 10;
}
